pybop/optimisers/base_optimiser.py

```diff
@@ -421,6 +427,7 @@ def add_result(self, result):
             time=result._time,  # noqa: SLF001
             scipy_result=result._scipy_result,  # noqa: SLF001
             x0=result._x0,  # noqa: SLF001
+            pybamm_solution=result.pybamm_solution,  # noqa: SLF001
```

Comment:
I don't believe we need to skip linting on this line:
```suggestion
            pybamm_solution=result.pybamm_solution,
```